cmd/internal/cli/inspect_linux.go

```diff
@@ -211,42 +223,113 @@ func getHelpCommand(appName string) string {
 	return getSingleFileCommand("runscript.help", "helpfile", appName)
 }
 
-func setAttribute(obj *inspectFormat, label string, value string) {
+func setAttribute(obj *inspectFormat, label, app string, value string) {
+	if app == "" {
+		app = "system-partition"
+	}
+
 	switch label {
 	case "apps":
-		obj.Attributes.Apps = value
+		obj.Data.Attributes.Apps = value
 	case "deffile":
-		obj.Attributes.Deffile = value
+		obj.Data.Attributes.Deffile = value
 	case "test":
-		obj.Attributes.Test = value
+		obj.Data.Attributes.Test = value
 	case "helpfile":
-		obj.Attributes.Helpfile = value
+		obj.Data.Attributes.Helpfile = value
 	case "labels":
-		if err := json.Unmarshal([]byte(value), &obj.Attributes.Labels); err != nil {
-			sylog.Warningf("Unable to parse labels: %s", value)
+		b, _, _, err := jsonparser.Get([]byte(value), app)
+		if err != nil {
+			sylog.Warningf("Unable to find json from metadata: %s; skipping", err)
+			b = []byte(value)
+		}
+
+		if err := json.Unmarshal(b, &obj.Data.Attributes.Labels); err != nil {
+			sylog.Warningf("Unable to parse labels: %s", err)
 		}
 	case "runscript":
-		obj.Attributes.Runscript = value
+		obj.Data.Attributes.Runscript = value
 	default:
 		if strings.HasSuffix(label, "environment.sh") {
-			obj.Attributes.Environment[label] = value
+			obj.Data.Attributes.Environment = value
 		} else {
 			sylog.Warningf("Trying to set attribute for unknown label: %s", label)
 		}
 	}
 }
 
-func getAppCheck(appName string) string {
-	return fmt.Sprintf("if ! [ -d \"/scif/apps/%s\" ]; then echo \"App %s does not exist.\"; exit 2; fi;", appName, appName)
-}
-
-// returns true if flags for other forms of information are unset
+// returns true if flags for other forms of information are unset.
 func defaultToLabels() bool {
 	return !(helpfile || deffile || runscript || testfile || environment || listApps)
 }
 
-// InspectCmd represents the 'inspect' command
-// TODO: This should be in its own package, not cli
+func inspectLabelPartition(inspectData *inspectFormat, fimg *sif.FileImage) error {
+	if fimg == nil {
```

Comment:
In sandbox labels are stored in `/.singularity.d/labels.json`. Why not using that?

Reply:
We do... _maybe I'm not understanding your comment..._

Reply:
Nevermind, my bad. You call exec for sandbox later.